src/random.cpp

```diff
@@ -156,14 +159,25 @@ void GetOSRand(unsigned char *ent32)
             RandFailure();
         }
     }
-#elif defined(HAVE_GETENTROPY)
+#elif defined(HAVE_GETENTROPY) || defined(HAVE_GETENTROPY_RAND)
     /* On OpenBSD this can return up to 256 bytes of entropy, will return an
      * error if more are requested.
      * The call cannot return less than the requested number of bytes.
      */
+    #if defined(MAC_OSX)
+    // Fallback for OSX < 10.12
+    if (&getentropy != NULL) {
```

Comment:
It seems harmless to do the `&getentropy != NULL` check for other platforms too (avoiding the MAC_OSX specific code).